### PR TITLE
Fixed bug related to empty environmental variable in a parameter command in ml_ops.sh

### PR DESCRIPTION
**Pull request by brandon-edwards**
Modified spot.conf to contain logic that only tries to pass the usedomain parameter when the spot.conf variable USER_DOMAIN is non-empty. I ran tests of the code with both empty and non-empty examples of USER_DOMAIN in spot.conf.

### CODE REVIEW

File: spot-ml/ml_ops.sh [hunk 1]
```diff
@@ -97,7 +105,7 @@ time spark-submit --class "org.apache.spot.SuspiciousConnects" \
   --dupfactor ${DUPFACTOR} \
   --feedback ${FEEDBACK_PATH} \
   --ldatopiccount ${TOPIC_COUNT} \
-  --userdomain ${USER_DOMAIN}\
+  $USER_DOMAIN_PARSER_CMD \
```
Comment (rabarona) [inline spot-ml/ml_ops.sh]: Probably this is not the first time we set an optional parameter. This approach is nice, I'd just separate fixed parameter from optional for cosmetics reasons, that means to leave all the "--" parameters first and at the end any optional parameter.

Comment (brandon-edwards) [reply spot-ml/ml_ops.sh]: Yes, suggested by Nathan. Ok, cool I'll make that change.

File: spot-ml/ml_ops.sh [hunk 2]
```diff
@@ -45,6 +45,14 @@ else
     RAWDATA_PATH=${PROXY_PATH}
 fi
 
+# pass the user domain designation if not empty
+
+if [ ! -z $USER_DOMAIN ] ; then
+    USER_DOMAIN_PARSER_CMD="--userdomain $USER_DOMAIN"
```
Comment (NathanSegerlind) [inline spot-ml/ml_ops.sh]: what's with the "PARSER" in the variable name? why not USER_DOMAIN_CMD or USER_DOMAIN_CLI_ENTRY ?

**Review by NathanSegerlind — APPROVED** [review]

**Review by rabarona — APPROVED** [review]

**Comment by brandon-edwards** [comment]
I see your point Nathan about the parsing piece of the command name. I changed that and also made another change related to a test I ran this morning - sorry I didn't run that test before committing. I'll be better about that in the future.

**Comment by brandon-edwards** [comment]
I just created a pull request to the incubator-spot repo which included these changes as well as the previous user domain designation changes.

**Comment by rabarona** [comment]
We still need to have a discussion on how we are going to push to incubator-spot or apache repo. As I see it needs to be a complete process with QA involved and per sprint pushes/planned releases.

**Comment by brandon-edwards** [comment]
Ok I resolved recent merge conflicts so that this can be merged - and did a test run. I cannot do it though, so one of you will have to. THX

**Pull request merged**